Source/JavaScriptCore/jit/RegisterSet.h

```diff
@@ -27,181 +27,456 @@
 
 #if !ENABLE(C_LOOP)
 
+#include "FPRInfo.h"
 #include "GPRInfo.h"
 #include "MacroAssembler.h"
 #include "Reg.h"
+#include "SimdInfo.h"
 #include <wtf/Bitmap.h>
+#include <wtf/CommaPrinter.h>
 
 namespace JSC {
 
+class SmallRegisterSet;
 typedef Bitmap<MacroAssembler::numGPRs + MacroAssembler::numFPRs> RegisterBitmap;
 class RegisterAtOffsetList;
+struct RegisterSetHash;
+class WholeRegisterSet;
+
+class RegisterSet final {
+    friend SmallRegisterSet;
+    friend WholeRegisterSet;
 
-class RegisterSet {
 public:
-    constexpr RegisterSet() = default;
+    constexpr RegisterSet() { }
+
+    inline constexpr RegisterSet(WholeRegisterSet);
 
     template<typename... Regs>
-    constexpr explicit RegisterSet(Regs... regs)
+    inline constexpr explicit RegisterSet(Regs... regs)
     {
         setMany(regs...);
     }
-    
-    JS_EXPORT_PRIVATE static RegisterSet stackRegisters();
-    JS_EXPORT_PRIVATE static RegisterSet reservedHardwareRegisters();
-    static RegisterSet runtimeTagRegisters();
-    static RegisterSet specialRegisters(); // The union of stack, reserved hardware, and runtime registers.
-    JS_EXPORT_PRIVATE static RegisterSet calleeSaveRegisters();
-    static RegisterSet vmCalleeSaveRegisters(); // Callee save registers that might be saved and used by any tier.
-    static RegisterAtOffsetList* vmCalleeSaveRegisterOffsets();
-    static RegisterSet llintBaselineCalleeSaveRegisters(); // Registers saved and used by the LLInt.
-    static RegisterSet dfgCalleeSaveRegisters(); // Registers saved and used by the DFG JIT.
-    static RegisterSet ftlCalleeSaveRegisters(); // Registers that might be saved and used by the FTL JIT.
-#if ENABLE(WEBASSEMBLY)
-    static RegisterSet webAssemblyCalleeSaveRegisters(); // Registers saved and used by the WebAssembly JIT.
-#endif
-    JS_EXPORT_PRIVATE static RegisterSet volatileRegistersForJSCall();
-    static RegisterSet stubUnavailableRegisters(); // The union of callee saves and special registers.
-    JS_EXPORT_PRIVATE static RegisterSet macroScratchRegisters();
-    JS_EXPORT_PRIVATE static RegisterSet allGPRs();
-    JS_EXPORT_PRIVATE static RegisterSet allFPRs();
-    static RegisterSet allRegisters();
-    JS_EXPORT_PRIVATE static RegisterSet argumentGPRS();
-
-    static RegisterSet registersToNotSaveForJSCall();
-    static RegisterSet registersToNotSaveForCCall();
-    
-    void set(Reg reg, bool value = true)
+
+    inline constexpr RegisterSet& includeRegister(Reg reg, Width width = Width128)
```

Comment:
I think these names look less unreadable than the previous name. For example, `contains` and `set` are better than `includesRegister` and `includeRegister`. Can you change that?
Since it is *Register*Set, I don't think we need to repeat `Register` in each function's name. `contains`, `set`, maybe `remove` is better.

Reply:
This was simply because those functions are overloaded in a bunch of places, and I wanted to be able to find all uses. We don't necessarily need to change them, but it made it easier to audit every use.

Reply:
I will undo this change, and make the width argument explicit